openml/study/study.py

```diff
@@ -1,13 +1,31 @@
 import collections
-import openml
+from typing import Dict, List, Optional
+
 import xmltodict
 
+import openml
+
 
-class OpenMLStudy(object):
+class BaseStudy(object):
 
-    def __init__(self, study_id, alias, main_entity_type, benchmark_suite,
-                 name, description, status, creation_date, creator, tags, data,
-                 tasks, flows, setups, runs):
+    def __init__(
+        self,
+        study_id: Optional[int],
+        alias: Optional[str],
+        main_entity_type: str,
+        benchmark_suite: Optional[int],
+        name: str,
+        description: str,
+        status: Optional[str],
+        creation_date: Optional[str],
+        creator: Optional[int],
+        tags: Optional[List[Dict]],
+        data: Optional[List[int]],
+        tasks: Optional[List[int]],
+        flows: Optional[List[int]],
+        runs: Optional[List[int]],
+        setups: Optional[List[int]],
+    ):
```

Comment:
optional parameters should have default values.
non-optional parameters should, in my opinion, be placed before optional ones.